core/geometry/src/test/geometry3d/OrderedRotationAngles.test.ts

```diff
@@ -130,55 +147,348 @@ describe("OrderedRotationAngles", () => {
+      0, 0, 1
+    );
+    ck.testMatrix3d(matrix, expectedMatrix);
+
+    // Three Rotations
+    x = Math.PI / 2, y = 1.16937, z = 0.0349066;  // 90, 67, and 2 degrees
+    // The three classic "base rotation matrixes" (columns-based from)
+    const rX = Matrix3d.createRowValues(
+      1, 0, 0,
+      0, Math.cos(x), -Math.sin(x),
+      0, Math.sin(x), Math.cos(x),
+    );
+    const rY = Matrix3d.createRowValues(
+      Math.cos(y), 0, Math.sin(y),
+      0, 1, 0,
+      -Math.sin(y), 0, Math.cos(y),
+    );
+    const rZ = Matrix3d.createRowValues(
+      Math.cos(z), -Math.sin(z), 0,
+      Math.sin(z), Math.cos(z), 0,
+      0, 0, 1,
+    );
+
+    /*
+    * Note: for the following checks the AxisOrder is the reverse
+    * of matrix multiplication because we used columns-based from
+    **/
+    // Rotation using XYZ ordering
+    OrderedRotationAngles.createRadians(x, y, z, AxisOrder.XYZ, angles);
+    angles.toMatrix3d(matrix);
+    expectedMatrix = rZ.multiplyMatrixMatrix(rY, expectedMatrix).multiplyMatrixMatrix(rX, expectedMatrix);
+    ck.testMatrix3d(matrix, expectedMatrix);
+
+    // Rotation using YXZ ordering
+    OrderedRotationAngles.createRadians(x, y, z, AxisOrder.YXZ, angles);
+    angles.toMatrix3d(matrix);
+    expectedMatrix = rZ.multiplyMatrixMatrix(rX, expectedMatrix).multiplyMatrixMatrix(rY, expectedMatrix);
+    ck.testMatrix3d(matrix, expectedMatrix);
+
+    // Rotation using ZXY ordering
+    OrderedRotationAngles.createRadians(x, y, z, AxisOrder.ZXY, angles);
+    angles.toMatrix3d(matrix);
+    expectedMatrix = rY.multiplyMatrixMatrix(rX, expectedMatrix).multiplyMatrixMatrix(rZ, expectedMatrix);
+    ck.testMatrix3d(matrix, expectedMatrix);
+
+    // Rotation using ZYX ordering
+    OrderedRotationAngles.createRadians(x, y, z, AxisOrder.ZYX, angles);
+    angles.toMatrix3d(matrix);
+    expectedMatrix = rX.multiplyMatrixMatrix(rY, expectedMatrix).multiplyMatrixMatrix(rZ, expectedMatrix);
+    ck.testMatrix3d(matrix, expectedMatrix);
+
+    // Rotation using YZX ordering
+    OrderedRotationAngles.createRadians(x, y, z, AxisOrder.YZX, angles);
+    angles.toMatrix3d(matrix);
+    expectedMatrix = rX.multiplyMatrixMatrix(rZ, expectedMatrix).multiplyMatrixMatrix(rY, expectedMatrix);
+    ck.testMatrix3d(matrix, expectedMatrix);
+
+    // Rotation using XZY ordering
+    OrderedRotationAngles.createRadians(x, y, z, AxisOrder.XZY, angles);
+    angles.toMatrix3d(matrix);
+    expectedMatrix = rY.multiplyMatrixMatrix(rZ, expectedMatrix).multiplyMatrixMatrix(rX, expectedMatrix);
+    ck.testMatrix3d(matrix, expectedMatrix);
+
+    expect(ck.getNumErrors()).equals(0);
+  });
+});
+
+describe("OrderedRotationAngles", () => {
+  it("OrderedRotationAngles.toMatrix3dRowBasedUsingClassicBaseMatrixes", () => {
+    const ck = new Checker();
+    let x = 0, y = 0, z = 0;
+    // we don't need the following line because default is false. It's added just for clarity.
+    OrderedRotationAngles.treatVectorsAsColumns = false;
```

Comment:
Actually, this line _is_ needed because this boolean is a global variable, and a previous test could've set it to true.

Reply:
will address in the next PR